django/db/migrations/autodetector.py

```diff
@@ -919,6 +939,46 @@ def generate_altered_fields(self):
                     self._generate_removed_field(app_label, model_name, field_name)
                     self._generate_added_field(app_label, model_name, field_name)
 
+    def create_altered_indexes(self):
+        option_name = operations.AddIndex.option_name
+        for app_label, model_name in sorted(self.kept_model_keys):
+            old_model_name = self.renamed_models.get((app_label, model_name), model_name)
+            old_model_state = self.from_state.models[app_label, old_model_name]
+            new_model_state = self.to_state.models[app_label, model_name]
+
+            old_indexes = old_model_state.options[option_name]
+            new_indexes = new_model_state.options[option_name]
```

Comment:
@MarkusH Just wanted to confirm if this is what you meant [here](https://github.com/django/django/pull/6857#discussion_r71702557) and `new_indexes` [cannot be](https://github.com/django/django/blob/283b468462a586fd9bf7a2794e9b9a20a7e8a2d9/django/db/migrations/state.py#L333) `None`.